### ISSUE TITLE
Running Tensorflow on CPU is faster than running it on GPU

### ISSUE THREAD

**Issue by rezaee** [tensorflow/tensorflow#31281]
I have an ASUS n552vw laptop that has a 4GB dedicated Geforce GTX 960M graphic card. I put these lines of code in the beginning of my code to compare training speed using GPU or CPU, and I saw it seems using the CPU wins!

For GPU:

    import os
    os.environ['CUDA_VISIBLE_DEVICES'] = '0'
    

For CPU:

    import os
    os.environ['CUDA_VISIBLE_DEVICES'] = '-1'
    

I have installed CUDA, cuDNN, tensorflow-gpu, etc to increase my training speed but seems inverse thing happened! 

When I try the first code, it says(before execution start):

    Train on 2128 samples, validate on 22 samples
    Epoch 1/1
    2019-08-02 18:49:41.828287: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX AVX2
    2019-08-02 18:49:42.457662: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1433] Found device 0 with properties: 
    name: GeForce GTX 960M major: 5 minor: 0 memoryClockRate(GHz): 1.176
    pciBusID: 0000:01:00.0
    totalMemory: 4.00GiB freeMemory: 3.34GiB
    2019-08-02 18:49:42.458819: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1512] Adding visible gpu devices: 0
    2019-08-02 18:49:43.776498: I tensorflow/core/common_runtime/gpu/gpu_device.cc:984] Device interconnect StreamExecutor with strength 1 edge matrix:
    2019-08-02 18:49:43.777007: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990]      0 
    2019-08-02 18:49:43.777385: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 0:   N 
    2019-08-02 18:49:43.777855: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 3050 MB memory) -> physical GPU (device: 0, name: GeForce GTX 960M, pci bus id: 0000:01:00.0, compute capability: 5.0)
    2019-08-02 18:49:51.834610: I tensorflow/stream_executor/dso_loader.cc:152] successfully opened CUDA library cublas64_100.dll locally

And it's really slow ` [Finished in 263.2s]`, But when I try the second code it says:

    Train on 2128 samples, validate on 22 samples
    Epoch 1/1
    2019-08-02 18:51:43.021867: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX AVX2
    2019-08-02 18:51:43.641123: E tensorflow/stream_executor/cuda/cuda_driver.cc:300] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected
    2019-08-02 18:51:43.645072: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:161] retrieving CUDA diagnostic information for host: DESKTOP-UQ8B9FK
    2019-08-02 18:51:43.645818: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:168] hostname: DESKTOP-UQ8B9FK

And it's much faster than the first code `[Finished in 104.7s]` ! How is it possible??

This is the part of code that is related to `Tensorflow` :

    model = Sequential()
    model.add((LSTM(un , return_sequences = True)))
    model.add(Dropout(dp)) 
    
    model.add((LSTM(un , return_sequences = True)))
    model.add(Dropout(dp)) 
    
    model.add((LSTM(un , return_sequences = True)))
    model.add(Dropout(dp)) 
    
    model.add((LSTM(un , return_sequences = True)))
    model.add(Dropout(dp)) 
    
    model.add((LSTM(un , return_sequences = False)))
    model.add(Dropout(dp)) 
    
    model.add(RepeatVector(rp))
      
    model.add((LSTM(un , return_sequences= True))) 
    model.add(Dropout(dp))   
    
    model.add((LSTM(un , return_sequences= True))) 
    model.add(Dropout(dp))
    
    model.add((LSTM(un , return_sequences= True))) 
    model.add(Dropout(dp))
    
    model.add((LSTM(un , return_sequences= True))) 
    model.add(Dropout(dp))
    
    model.add((LSTM(un , return_sequences= True))) 
    model.add(Dropout(dp))

    model.add(TimeDistributed(Dense(ds)))

**Comment by forReason** [comment]
Im getting a 60% performance "boost" without GPU!
Im training on 2 different systems.  
My server, without a GPU:
- Intel Core i5 6500T (4x@2.5 ghz) Notebook processor
- Network attached storage for training data and output
- 16 Gb DDR 3 Ram

My Desktop PC:
- Intel core i7 3770k (4x3,5-4 ghz)
- Nvidia GTX 970
- 32gb ddr3 ram
- Same training data on NAS

Now interestingly 3000 epochs, 100000 records each takes roughly 3h on the server using TF 1.14
The same on my Desktop with GPU takes 8h with TF 2.0 
It clearly doesn't utilize my GPU properly. It sits with full Ram but at 3% graphical processor use. 
The main processor is at 30% use while 100% with any CPU build.

My model is a fairly simple keras sequential lstm:

```
for learningrate in learningrates:
	for layerdensity in layerdensitys:
		for layer in amount_of_layers:
			################################
			# generate model               #
			################################
			modelname = f"{layer}-layer_{layerdensity}-nodes_selu-adam_{learningrate}-learningrate_{records_per_epoch}-epochsize_{appendix}"
			model = keras.Sequential()
			## layertypes ##
			# dense // general
			# convolutional; max pooling // image classification
			# CuDNNLSTM // long short term memory (cudnn gpu optimized otherwise just lstm) (same as dense just cudnn)
			model.add(Dense(layerdensity, activation=tf.nn.selu, input_dim=15))
			for i in range(layer-1):
				model.add(Dense(layerdensity, activation=tf.nn.selu))
			model.add(Dense(9,activation=tf.nn.softmax, name = "Output"))
			# Compile
			optimizer = tf.keras.optimizers.Adam(lr=learningrate)
			model.compile(
				optimizer=optimizer,
				loss='sparse_categorical_crossentropy',
				metrics=['accuracy'])
			model.summary()
			tensorboard = TensorBoard(log_dir="\\\\drg-fs01\\BigData\\Projects\\Notebooks\\PokerBot\\log\\" + modelname,
				histogram_freq = 100, write_graph = False)
            #cp_callback = tf.keras.callbacks.ModelCheckpoint("\\\\drg-fs01\\BigData\\Projects\\Notebooks\\PokerBot\\checkpoints\\" + modelname, verbose=0)
            ################################
            # train model                  #
            ################################
			model.fit(trainSet, 
				epochs = epochs, 
				steps_per_epoch = trainSteps, 
				shuffle = True, 
				validation_data = testSet, 
				validation_steps = testSteps, 
				validation_freq = int(epochs/maxTestEpochs),
				verbose = verbose, 
				callbacks = [tensorboard])#,cp_callback])
			model.save(basePath+'saved_models/' + modelname + '.h5')
```
How can I debug this?

**Comment by gadagashwini** [comment]
@rezaee Please provide details about what platform you are using (operating system, architecture). Also include your TensorFlow version. Also, did you compile from source or install a binary?
In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!

**Comment by gadagashwini** [comment]
@forReason Can you please post a new issue by providing the information asked by the [template](https://github.com/tensorflow/tensorflow/issues/new/choose)?
The reason for this is we can focus on your specific configuration and problem since the root cause can be unrelated even though the error messages are similar. Thanks!

**Comment by gadagashwini** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
